src/AMSlib/wf/workflow.hpp

```diff
@@ -149,50 +161,59 @@ class AMSWorkflow
   AMSWorkflow()
       : AppCall(nullptr),
         DB(nullptr),
-        dbType(AMSDBType::None),
-        appDataLoc(AMSResourceType::HOST),
-        ePolicy(AMSExecPolicy::UBALANCED)
-  {
-#ifdef __ENABLE_DB__
-    auto &dbm = ams::DBManager<FPTypeValue>::getInstance();
-    DB = dbm.createDB("miniApp_data.txt", dbType, 0);
-    CFATAL(WORKFLOW, !DB, "Cannot create database");
+        appDataLoc(AMSResourceType::AMS_HOST),
+#ifdef __AMS_ENABLE_MPI__
+        comm(MPI_COMM_NULL),
 #endif
+        ePolicy(AMSExecPolicy::AMS_UBALANCED)
+  {
   }
 
   AMSWorkflow(AMSPhysicFn _AppCall,
-              char *uq_path,
-              char *surrogate_path,
-              char *db_path,
-              const AMSDBType dbType,
-              AMSResourceType appDataLoc,
+              std::string &uq_path,
+              std::string &surrogate_path,
+              std::string &domain_name,
+              AMSResourceType app_data_loc,
               FPTypeValue threshold,
-              const AMSUQPolicy uqPolicy,
+              const AMSUQPolicy uq_policy,
               const int nClusters,
               int _pId = 0,
-              int _wSize = 1,
-              AMSExecPolicy policy = AMSExecPolicy::UBALANCED)
+              int _wSize = 1)
       : AppCall(_AppCall),
-        dbType(dbType),
+        domainName(domain_name),
         rId(_pId),
         wSize(_wSize),
-        appDataLoc(appDataLoc),
-        uqPolicy(uqPolicy),
-        ePolicy(policy)
+        appDataLoc(app_data_loc),
+        uqPolicy(uq_policy),
+#ifdef __AMS_ENABLE_MPI__
+        comm(MPI_COMM_NULL),
+#endif
+        ePolicy(AMSExecPolicy::AMS_UBALANCED)
+
   {
     DB = nullptr;
-    if (db_path) {
-      DBG(Workflow, "Creating Database");
-      auto &dbm = ams::DBManager<FPTypeValue>::getInstance();
-      DB = dbm.getDB(db_path, dbType, rId);
-    }
+    auto &dbm = ams::db::DBManager::getInstance();
 
+    DB = dbm.getDB(domainName, rId);
     UQModel = std::make_unique<UQ<FPTypeValue>>(
         appDataLoc, uqPolicy, uq_path, nClusters, surrogate_path, threshold);
   }
 
   void set_physics(AMSPhysicFn _AppCall) { AppCall = _AppCall; }
 
+  void set_communicator(MPI_Comm communicator) { comm = communicator; }
```

Comment:
I'm not sure what is the intended usage. If it's to create a new communicator based on an existing one we should do MPI_Comm_dup

Reply:
For now this a place holder. The MPI/ load balance has a bug.